Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -26,5 +26,29 @@ public async Task<IEnumerable<JokeReplyDto>> GetAllJokeDtosAsync()
 
             return jokeDtos;
         }
+
+        public async Task UpdateJoke(JokeUpdateDTO jokeUpdateDTO)
+        {
+            // get the joke we need to update
+            // set fields with jokeUpdateDTO
+            var jokeToUpdate = await _jokesRepo.GetJokeToUpdate(jokeUpdateDTO);
+
+            if (jokeToUpdate == null)
+            {
+                throw new NotImplementedException("can't find joke to update");
```

Comment:
Instead of throwing this exception here I think you should return a NotFound() result in the controller.

Reply:
use `KeyNotFoundException` then the GlobalExeptionhandler will handle returning it.
 [GlobalExeptionHandler](https://github.com/Code2Gether-Discord/JokesOnYou/blob/dev/Server/JokesOnYou.Web.Api/Middlewares/ExceptionMiddleware.cs)